java-checks/src/main/java/org/sonar/java/checks/UndocumentedApiCheck.java

```diff
@@ -367,23 +396,23 @@ private static boolean isEmptyDescription(@Nullable String part) {
     private static List<String> getExceptions(Tree tree) {
       if (tree.is(METHOD_KINDS)) {
         return ((MethodTree) tree).throwsClauses().stream()
-          .map(Javadoc::toIdentifier)
+          .map(Javadoc::exceptionName)
           .filter(Objects::nonNull)
-          .map(IdentifierTree::name)
           .collect(Collectors.toList());
       }
       return Collections.emptyList();
     }
 
-    private static IdentifierTree toIdentifier(TypeTree typeTree) {
+    private static String exceptionName(TypeTree typeTree) {
       switch (typeTree.kind()) {
         case IDENTIFIER:
-          return (IdentifierTree) typeTree;
+          return ((IdentifierTree) typeTree).name();
         case MEMBER_SELECT:
-          return ((MemberSelectExpressionTree) typeTree).identifier();
+          return ExpressionsHelper.concatenate((MemberSelectExpressionTree) typeTree);
         default:
-          throw new IllegalStateException(
-            "Should never happen. Exceptions can not be specified other than with an identifier or a fully qualified name, and can not be parametrized.");
+          // should not happen, as exceptions can only be defined using identifier or fully qualified name,
+          // no parameterized type can be used
+          return null;
```

Comment:
So why actually returning something ? I would really prefer to stick to throwing an explicit exception. AFAI understand you only return so you can test it in an existing test : create a dedicated test for the case and keep the explicit failure.